### DISCUSSION TITLE
"No Bluetooth devies recorded in selected time frame"

### DISCUSSION THREAD

**Discussion by sudo-joe** [nzymedefense/nzyme#1146] [Questions] [answered]
Hello

-> Dashboard -> Bluetooth -> Devices ->
"No Bluetooth devies recorded in selected time frame"

Before starting Nzmye, I did "setup" Bluetooth by

'''
ubuntu@ubuntu:~/Desktop$ sudo hciconfig
hci0:	Type: Primary  Bus: USB
	BD Address: F4:bb:cc:dd:ee:ff  ACL MTU: 1022:8  SCO MTU: 183:5
	UP RUNNING 
	RX bytes:1372 acl:0 sco:0 events:97 errors:0
	TX bytes:5792 acl:0 sco:0 commands:97 errors:0

ubuntu@ubuntu:~/Desktop$ sudo hciconfig hci0 up
ubuntu@ubuntu:~/Desktop$ 
'''

2.
How should a prpoerly configured nzyme-tap.conf look like if the interface in question should be enabled?

[Like that?]
[bluetooth_interfaces.hci0]
active = true
bt_classic_enabled = true
bt_le_enabled = true
discovery_period_seconds = 15
dbus_method_call_timeout_seconds = 2

[Or like that?]
bluetooth_interfaces.hci0
active = true
bt_classic_enabled = true
bt_le_enabled = true
discovery_period_seconds = 15
dbus_method_call_timeout_seconds = 2

PS
I am on [2.0.0-alpha.14](https://github.com/nzymedefense/nzyme/releases/tag/2.0.0-alpha.14)

**Comment by lennartkoopmann** [comment]
Do you see any errors in the `nzyme-tap` logs? It will not start up if you use the wrong configuration syntax. The correct syntax is always in the example configuration file and, in general, it is TOML formatted.

**Comment by sudo-joe** [comment]
Hmm....

I get something like an error when terminating gedit:
** (gedit:10373): WARNING **: 06:20:25.475: Set document metadata failed: Setting attribute metadata::gedit-position not supported


2.
->Do you see any errors in the nzyme-tap logs?

No, or do i miss something?

ubuntu@ubuntu:~/Desktop$ sudo tail -f /var/log/nzyme/nzyme.log
2024-09-11T07:18:50.252+0000 [periodicals-3] INFO  app.nzyme.core.periodicals.versioncheck.VersioncheckThread - Starting to check for most recent nzyme version.
2024-09-11T07:18:50.262+0000 [main] INFO  app.nzyme.core.plugin.loading.PluginLoader - Scanning [/usr/share/nzyme/plugins] for plugins.
2024-09-11T07:18:50.268+0000 [main] WARN  app.nzyme.core.plugin.loading.PluginLoader - File [/usr/share/nzyme/plugins/.gitkeep] in plugins folder does not have .jar suffix. Skipping.
2024-09-11T07:18:52.070+0000 [periodicals-3] INFO  app.nzyme.core.periodicals.versioncheck.VersioncheckThread - Successfully completed version check. Everything seems up to date.
2024-09-11T07:18:53.528+0000 [main] INFO  app.nzyme.core.rest.server.NzymeHttpServer - Loaded TLS certificate: [GENERATED_SELF_SIGNED/723f4f0df3a6ad87c9d57a0741fce9ddb0cd68f4681eafef3a06fbab01373ad3]
2024-09-11T07:18:53.671+0000 [main] INFO  app.nzyme.core.rest.server.NzymeHttpServer - Started web interface and REST API at [https://127.0.0.1:22900/]. Access it at: [https://127.0.0.1:22900/]
2024-09-11T07:18:53.997+0000 [registry-change-processor-0] INFO  app.nzyme.core.registry.RegistryChangeMonitorImpl - Registry key [core.versioncheck_status] now has a value. Notifying subscribers.
2024-09-11T07:19:52.756+0000 [grizzly-http-server-1] INFO  app.nzyme.core.rest.resources.system.authentication.mgmt.InitialUserResource - Creating initial user [banditcatcher@abcdef.net].
2024-09-11T07:20:09.018+0000 [grizzly-http-server-4] INFO  app.nzyme.core.rest.resources.system.authentication.AuthenticationResource - Creating session for user [banditcatcher@abcdef.net]
2024-09-11T07:20:56.266+0000 [grizzly-http-server-2] INFO  app.nzyme.core.rest.resources.system.authentication.AuthenticationResource - User [banditcatcher@abcdef.net] passed MFA challenge with recovery code.

**Comment by sudo-joe** [comment]
->Do you see any errors in the nzyme-tap logs?
No, there is not even an nzyme-tap file in /var/log/nzyme :(

**Comment by lennartkoopmann** [comment]
The `nzyme-tap` logs are not written to disk but available in `journalctl`

**Comment by sudo-joe** [comment]
```
● nzyme-tap.service - Nzyme-Tap
     Loaded: loaded (/lib/systemd/system/nzyme-tap.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2024-10-02 08:05:34 UTC; 8s ago
       Docs: https://github.com/lennartkoopmann/nzyme
   Main PID: 17448 (nzyme-tap)
      Tasks: 21 (limit: 9094)
     Memory: 14.7M
        CPU: 128ms
     CGroup: /system.slice/nzyme-tap.service
             └─17448 /usr/bin/nzyme-tap --configuration-file /etc/nzyme/nzyme-tap.conf --log-level info

Oct 02 08:05:34 ubuntu nzyme-tap[17448]: [INFO][2024-10-02 08:05:34][ethernet::capture] Available network device: nfqueue (Linux n>
Oct 02 08:05:34 ubuntu nzyme-tap[17448]: [INFO][2024-10-02 08:05:34][ethernet::capture] Available network device: dbus-system (D-B>
Oct 02 08:05:34 ubuntu nzyme-tap[17448]: [INFO][2024-10-02 08:05:34][ethernet::capture] Available network device: dbus-session (D->
Oct 02 08:05:34 ubuntu nzyme-tap[17448]: [INFO][2024-10-02 08:05:34][ethernet::ethernet_broker] Installing ethernet broker thread >
Oct 02 08:05:34 ubuntu nzyme-tap[17448]: [INFO][2024-10-02 08:05:34][dot11::dot11_broker] Installing WiFi broker thread <0>.
Oct 02 08:05:34 ubuntu nzyme-tap[17448]: [INFO][2024-10-02 08:05:34][ethernet::capture] Starting ethernet capture on [enp3s0f2].
Oct 02 08:05:34 ubuntu nzyme-tap[17448]: [INFO][2024-10-02 08:05:34][bluetooth::capture] Starting Bluetooth capture on [hci0]
lines 1-18


-----------------------------------------------------------------------------------------------------------------------------------

ubuntu@ubuntu:~$ journalctl -xe
░░ The job identifier is 1101.
Oct 02 08:10:24 ubuntu gnome-shell[19123]: # Couldn't connect to accessibility bus: Failed to connect to socket />
Oct 02 08:10:24 ubuntu systemd[1674]: Started VTE child process 19127 launched by gnome-terminal-server process 6>
░░ Subject: A start job for unit UNIT has finished successfully
░░ Defined-By: systemd
░░ Support: http://www.ubuntu.com/support
░░ 
░░ A start job for unit UNIT has finished successfully.
░░ 
░░ The job identifier is 1105.
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][link::leaderlink] Could not send status. er>
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][nzyme_tap] Node hello failed. Retrying. Err>
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][dot11::tables::dot11_table] Could not submi>
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][bluetooth::tables::bluetooth_table] Could n>
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][ethernet::tables::tcp_table] Could not subm>
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][ethernet::tables::udp_table] Could not subm>
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][ethernet::tables::dns_table] Could not subm>
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][ethernet::tables::ssh_table] Could not subm>
Oct 02 08:10:25 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:25][ethernet::tables::socks_table] Could not su>
Oct 02 08:10:28 ubuntu kernel: workqueue: delayed_fput hogged CPU for >10000us 8 times, consider switching to WQ_>
Oct 02 08:10:30 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:30][nzyme_tap] Node hello failed. Retrying. Err>
Oct 02 08:10:34 ubuntu nzyme-tap[17448]: [INFO][2024-10-02 08:10:34][system_state] Training period ended after <5>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][link::leaderlink] Could not send status. er>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][nzyme_tap] Node hello failed. Retrying. Err>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][dot11::tables::dot11_table] Could not submi>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][bluetooth::tables::bluetooth_table] Could n>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][ethernet::tables::tcp_table] Could not subm>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][ethernet::tables::udp_table] Could not subm>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][ethernet::tables::dns_table] Could not subm>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][ethernet::tables::ssh_table] Could not subm>
Oct 02 08:10:36 ubuntu nzyme-tap[17448]: [ERROR][2024-10-02 08:10:36][ethernet::tables::socks_table] Could not su>
lines 1215-1245/1245 (END)


-----------------------------------------------------------------------------------------------------------------------------------

● nzyme.service - Nzyme
     Loaded: loaded (/lib/systemd/system/nzyme.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2024-10-02 07:40:47 UTC; 24s ago
       Docs: https://github.com/lennartkoopmann/nzyme
   Main PID: 10370 (nzyme)
      Tasks: 62 (limit: 9094)
     Memory: 769.4M
        CPU: 48.373s
     CGroup: /system.slice/nzyme.service
             ├─10370 /bin/sh /usr/share/nzyme/bin/nzyme
             └─10371 /usr/bin/java -Xms1g -Xmx1g -XX:-OmitStackTraceInFastThrow -jar -Dlog4j.configurationFile=file:///etc/nzyme/>

Oct 02 07:40:58 ubuntu nzyme[10371]: INFO: Update command completed successfully.
Oct 02 07:40:58 ubuntu nzyme[10371]: Liquibase: Update has been successful.
```

**Comment by sudo-joe** [comment, marked as answer]
Sorry for the noise I made. The problem was a configuration-error in nzyme-tap.conf.
